### DISCUSSION TITLE
Error while deploying Database in HONO, using connection pool (prisma accelerate)

### DISCUSSION THREAD

**Discussion by tb221** [prisma/prisma#24157] [Q&A]
### Question

I am getting this error while deploying hte databse code in HONO and connecting it with PRISMA accelerate.
NOTE : I am using neon db

![image](https://github.com/prisma/prisma/assets/61058735/9f4d0ac2-a963-4ab5-9eb8-98246e309ab7)

### Expected behavior (optional)

This is my code
```js
import { PrismaClient } from '@prisma/client/edge'
import { withAccelerate } from '@prisma/extension-accelerate'
import { Hono } from 'hono';
import { sign } from 'hono/jwt'

// Create the main Hono app
const app = new Hono<{
	Bindings: {
		DATABASE_URL: string
    JWT_SECRET: string
	}
}>();

app.post('/api/v1/signup', async (c,next) => {
	const prisma = new PrismaClient({
		datasourceUrl: c.env?.DATABASE_URL	,
	}).$extends(withAccelerate());

	const body = await c.req.json();
  console.log(body.email);
  console.log(body.password);

	try {
		const user = await prisma.user.create({
			data: {
				email: body.email,
				password: body.password
			}
		});
		const jwt = await sign({ id: user.id }, c.env.JWT_SECRET);
		return c.json({ jwt });
    
	} catch(e) {
		c.status(403);
		return c.json({ error: e});
	}
})

export default app;
```

and my schema is
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id       Int      @id @default(autoincrement())
  email    String   @unique
  name     String?
  password String
  posts    Post[]
}

model Post {
  id        Int      @id @default(autoincrement())
  title     String
  content   String
  published Boolean  @default(false)
  author    User     @relation(fields: [authorId], references: [id])
  authorId  Int
}
```

NOTE : First I migrate the db using the key provided by neon db and then I used the key to generate prisma accelerator key that is what I am using in wrangler.toml as an environment variable

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by tb221** [comment]
Prisma Version is 10.7.0

**Comment by ludralph** [comment]
Hi @tb221 👋 

Thank you for raising this question.

From the error message you shared, it is not clear what the issue is, Do you get any other error code in the range of [P6XXX](https://www.prisma.io/docs/orm/reference/error-reference#prisma-accelerate)?

**Reply by tb221** [reply]
Hi @ludralph ,
No, The error code that I got is P5010.
I tried connection pooling but I was not able to store the data into database

**Reply by Paper-Bag-dev** [reply]
Did you figure out any solution for this issue? I am facing the same issue atm with hono and aivencloud

**Reply by ludralph** [reply]
Hi @tb221 

In your Prisma schema, you need to enable the driverAdapters preview feature. Update your generator block to:
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["driverAdapters"]
}
```
1. Install the necessary packages:
```
npm install @prisma/adapter-neon @neondatabase/serverless
```

2. Modify your Prisma Client initialization to use the Neon adapter:
```js
import { PrismaClient } from '@prisma/client/edge'
import { withAccelerate } from '@prisma/extension-accelerate'
import { Pool } from '@neondatabase/serverless'
import { PrismaNeon } from '@prisma/adapter-neon'

// ... rest of your imports

app.post('/api/v1/signup', async (c, next) => {
  const pool = new Pool({ connectionString: c.env.DATABASE_URL })
  const adapter = new PrismaNeon(pool)
  const prisma = new PrismaClient({ adapter }).$extends(withAccelerate())

  // ... rest of your code
})
```

3. Make sure your `DATABASE_URL` in your wrangler.toml or environment variables is the Prisma Accelerate URL, which should look like:
```
DATABASE_URL="prisma://accelerate.prisma-data.net/?api_key=YOUR_API_KEY"
```

4. Run `npx prisma generate` to update your Prisma Client.

These changes should allow Prisma to work correctly in your Cloudflare Worker environment with Neon DB and Prisma Accelerate.
https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-cloudflare#neon

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
